projects/testing-library/src/lib/models.ts

```diff
@@ -1,8 +1,14 @@
-import { Type, DebugElement } from '@angular/core';
-import {ComponentFixture, DeferBlockBehavior, DeferBlockState, TestBed} from '@angular/core/testing';
+import { Type, DebugElement, OutputRef } from '@angular/core';
+import { ComponentFixture, DeferBlockBehavior, DeferBlockState, TestBed } from '@angular/core/testing';
 import { Routes } from '@angular/router';
 import { BoundFunction, Queries, queries, Config as dtlConfig, PrettyDOMOptions } from '@testing-library/dom';
 
+export type OutputRefKeysWithCallback<T> = {
+  [key in keyof T as T[key] extends OutputRef<any> ? key : never]?: T[key] extends OutputRef<infer U>
+    ? (val: U) => void
+    : never;
+};
+
```

Comment:
Using this results that the callback in the test always receives `any` as input argument.
I think this should be the following (`EventEmitter` needs to be imported).

The rest looks awesome!

 ```suggestion
export type OutputRefKeysWithCallback<T> = {
  [key in keyof T]: T[key] extends EventEmitter<infer U>
    ? (val: U) => void
    : T[key] extends OutputRef<infer U> 
    ?  (val: U) => void
    : never;
};

```

Reply:
Good Catch! 

I added a test to ensure there's no `any` in it. Not sure whether you want such noop / type only tests. I will remove otherwise - just let me know.

Reply:
That's fine, thanks!